### ISSUE TITLE
Tune the background fetcher

### ISSUE THREAD

**Issue by Kern--** [awslabs/soci-snapshotter#225]
**Is your feature request related to a problem? Please describe.**
https://github.com/awslabs/soci-snapshotter/pull/214 adds a background fetcher that will download spans over time, even if they are not requested by a synchronous file read. The fetcher has a few tunable parameters:
```
// SilencePeriodMSec defines the time (in ms) the background fetcher
// will be paused for when a new image is mounted.
SilencePeriodMSec int64 `toml:"silence_period_msec"`

// FetchPeriodMSec specifies how often a background fetch will occur.
// The background fetcher will fetch one span every FetchPeriodMSec.
FetchPeriodMSec int64 `toml:"fetch_period_msec"`
```

These are set arbitrarily to try to balance downloading all the data quickly to reduce the dependency on the network and not downloading so quickly that it competes with synchronous accesses in terms of CPU/Network/SOCI lock contention. 

**Describe the solution you'd like**
We should run some experiments while changing these parameters to see how it affects the overall performance of the snapshotter. The benchmarking framework might be a good place for this. The outcome should be a reasonable set of deafult values/recommendations for these parameters.

**Describe alternatives you've considered**
This type of tuning might be dependent on how exactly the snapshotter is being used across multiple containers and so our tuning could end up being overly aggressive in a real world scenario. It might only make sense to do this sort of tuning by the consumers of the snapshotter.

**Additional context**
N/A

**Comment by kzys** [comment]
It would be great the snapshotter can decide the values. It knows how busy the FUSE part is and has access to host-level information such as CPU utilization.

**Comment by Kern--** [comment]
I think you're probably right, but designing and implementing a dynamic background fetcher seems like a much larger task. I think this is decent intermediate goal. What do you think?

**Comment by kzys** [comment]
Yeah. We should pick okay-enough defaults. But we shouldn't try too hard to find the best defaults.

**Comment by wmesard** [comment]
**TL;DR:** +1.

Performance is a non-goal for the background fetcher; it is an availability feature.  It exists to achieve one thing: To close the window where a new container failure mode is possible (due to a network issue, or other problem that could cause lazy loading to fail). As such, background fetching should be very conservative, since the priority is to make sure that it does not interfere with operations that *are* performance related.

To echo what @kzys said, the background fetcher is in the best position to know what else is going on in the system, and to throttle itself accordingly. That's why it's implemented as a singleton. And to echo what @Kern-- said, that's a harder problem than tuning some default. But it is the right problem to solve.
